snap/component.go

```diff
@@ -33,6 +33,9 @@ type ComponentInfo struct {
 	Version     string              `yaml:"version"`
 	Summary     string              `yaml:"summary"`
 	Description string              `yaml:"description"`
+
+	// TODO: we will need to add fields here to carry around details about
+	// explicit and implicit hooks.
```

Comment:
ComponentInfo itself might need to carry copies of Plugs and Hook information. One approach to all of this could be to give a Bind(snap.Info....) method to ComponentInfo perhaps and a Bound() bool predicate? ComponentInfo would complete itself with bits of Info from the main snap when need by invoking this?

Reply:
it feels that at the point when we address this we could merge Component and ComponentInfo though? (as now we have the internal componentYaml to do what before Component did) ?